components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -355,48 +356,30 @@ class GeckoEngineSession(
 
             val response = settings.requestInterceptor?.onLoadRequest(
                 this@GeckoEngineSession,
-                request.uri
+                request.uri,
+                request.hasUserGesture,
+                this@GeckoEngineSession.currentUrl?.tryGetHostFromUrl() == request.uri.tryGetHostFromUrl()
             )?.apply {
                 when (this) {
                     is InterceptionResponse.Content -> loadData(data, mimeType, encoding)
                     is InterceptionResponse.Url -> loadUrl(url)
+                    is InterceptionResponse.AppIntent -> {
+                        notifyObservers {
```

Comment:
That's a problem / breaking change, as you're now only notifying the observer for app intents. Fenix has an observer too: https://github.com/mozilla-mobile/fenix/blob/fe034226a335a2e48b4fa94e94aa80d7fe278f57/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt#L401

So, it would no longer be called now. Can you just notify in all cases with the app intent being null for regular load requests?

Reply:
Understood.  I'll fix this.